### ISSUE TITLE
clang-tidy test failures in darwin builds

### ISSUE THREAD

**Issue by hiraditya** [llvm/llvm-project#60304]
Summary:

This error happens when a fat binary (arm64 + x86_64) is built.

When clang-tidy doesn't find a compilation command for a particular file (trivially-destructible.cpp) it tries to infer the command, and in this case it picks up the compilation command of a file (FasterStringFindCheck.cpp) from a parent directory.  clang-tidy cleans up the build command a little to make it suitable for `-cc1` invocation. The problem is that clang-tidy doesn't cleanup the **-arch** flag, and because of that there are two `cc1` invocations, one for each `arm64` and `x86_64`.

Thus build fails with "error: unable to handle compilation, expected exactly one compiler job".

Shortened error message:

> error: unable to handle compilation, expected exactly one compiler job in
./bin/clang++" "-cc1" "-triple" "arm64-apple-macosx10.14.0 ...... \
   tools/clang/tools/extra/test/clang-tidy/checkers/performance/Output/trivially-destructible.cpp.tmp.cpp ; \ <------------ second command in the same invocation.
  ./bin/clang++" "-cc1" "-triple" "x86_64-apple-macosx10.14.0" ........ \
  tools/clang/tools/extra/test/clang-tidy/checkers/performance/Output/trivially-destructible.cpp.tmp.cpp;

Repro steps:
```console
$ cmake -GNinja ../llvm -DLLVM_ENABLE_PROJECTS='clang;clang-tools-extra' -DCMAKE_BUILD_TYPE=RelWithDebInfo -DLLVM_TARGETS_TO_BUILD='AArch64;X86' -DLLVM_BUILD_LLVM_DYLIB=ON -DCMAKE_OSX_ARCHITECTURES='arm64;x86_64' -DCMAKE_OSX_DEPLOYMENT_TARGET=10.14

$ ninja check-clang-tools
```

Failed Tests (1):
  Clang Tools :: clang-tidy/checkers/performance/trivially-destructible.cpp

**Comment by hiraditya** [comment]
[Related issue](https://bugs.llvm.org/show_bug.cgi?id=46644)

**Comment by hiraditya** [comment]
So clang-tidy, or maybe clang-tooling?, infers compilation commands when it can't find the command for the file it is processing. In this case clang-tidy is using compilation command of `FasterStringFindCheck.cpp`


```
* thread #1, queue = 'com.apple.main-thread', stop reason = step over
    frame #0: 0x00000001006f5870 clang-tidy`clang::tooling::ClangTool::run(this=0x000000016fdfe8c0, Action=0x000000016fdfe358) at Tooling.cpp:538:32 [opt]
   535 	    // requirements to the order of invocation of its members.
   536 	    std::vector<CompileCommand> CompileCommandsForFile =
   537 	        Compilations.getCompileCommands(File);
-> 538 	    if (CompileCommandsForFile.empty()) {
   539 	      llvm::errs() << "Skipping " << File << ". Compile command not found.\n";
   540 	      FileSkipped = true;
   541 	      continue;
Target 0: (clang-tidy) stopped.
(lldb) p CompileCommandsForFile
(std::vector<clang::tooling::CompileCommand, std::allocator<clang::tooling::CompileCommand> >) $1 = size=1 {
  [0] = {
    Directory = "/"
    Filename = "llvm-project/build/tools/clang/tools/extra/test/clang-tidy/checkers/performance/Output/trivially-destructible.cpp.tmp.cpp"
    CommandLine = size=56 {
      [0] = "/Library/Developer/CommandLineTools/usr/bin/c++"

      [43] = "-arch"
      [44] = "arm64"
      [45] = "-arch"
      [46] = "x86_64"
     ...
     [55] = "/llvm-project/build/tools/clang/tools/extra/test/clang-tidy/checkers/performance/Output/trivially-destructible.cpp.tmp.cpp"
    }
    Output = ""
    Heuristic = "inferred from llvm-project/clang-tools-extra/clang-tidy/performance/FasterStringFindCheck.cpp"
  }

```

**Comment by hiraditya** [comment]
Fixed in 626849c71e85d546a004cc91866beab610222194
